### ISSUE TITLE
if  subscripts and supscripts are required at the same time

### ISSUE THREAD

**Issue by Chemprogrammer** [gebeleysis/RCTagProcessing#2]
example,  X<sup>3</sup><sub style=" margin:0 0 0 -5px">3</sub> , what should I do? Thanks!

**Comment by gebeleysis** [comment]
Hello,
I don't think there's currently a way to do what you want using `NSAttributedString`, hence no way to do it with the RCTagProcessor either.
What I would suggest is decide what needs to come first. E.g. in your example I think the subscript 3 is what's intrinsic to the X variable, it's "the third X" whereas the superscript 3 is the power it's being raised to, so in this particular example I'd write them the other way around, first sub, then sup.

That being said, mathematical notations can be quite complex and there can be cases where you'd actually want the subscript and superscript to be horizontally aligned. For example, structures like "sum from <> to <> of <>" like here:
![Image of sum expression](https://wikimedia.org/api/rest_v1/media/math/render/svg/6d4c7ece6af8eed0a4e3167c255d04f0b7bca83b)
This kind of structures are a struggle even for full-blown HTML+CSS. So much so, that Wikipedia renders them as images (using some kind of server-side tool I'd wager).

The purpose of this particular class was to make an equivalent of the abilities built into Android labels, as such, I would have to say that this edge-case is probably out of scope for the current class.

However, if you find a way to do this via `NSAttributedString`, in an Android label, or with some simple HTML tag, I'd be glad to be proven wrong. Keep in touch.

**Comment by Chemprogrammer** [comment]
To display X<sup>3</sup><sub style=" margin:0 0 0 -5px;">3</sub>, I use
UITextView *text.attributedText=[[ NSMutableAttributedString alloc] initWithData:[string dataUsingEncoding:NSUnicodeStringEncoding] options:@{ NSDocumentTypeDocumentAttribute: NSHTMLTextDocumentType } documentAttributes:nil error:nil];
My solution only works for UITextView, but it does not work for UILable or UIButton title. So I find your project, RCTagProcessor works perfectly for  subscript or superscript. However, when  subscript and superscript are required at the same time, it does not work.
Keep in touch,  many thanks!

**Comment by gebeleysis** [comment]
My understanding was that you want to have the two threes, subscript and superscript be left-aligned. Meaning start right after the X.
I think I may have misunderstood you. What is the value of `string` in your example?
Here's what I get when I use the html string you used in your comment
`NSString *string = @"X<sup>3</sup><sub style=\" margin:0 0 0 -5px;\">3</sub>";`
![screen shot 2017-07-12 at 10 34 09](https://user-images.githubusercontent.com/9152593/28111569-debc331c-66ed-11e7-9211-cd3494361faa.png)
This is from the demo app and the UITextView is the bottom-most element. The other two labels and a button are using `"X<sup>3</sup><sub>3</sub>"`.

In my testing I wasn't able to get any use out of the `style=" margin:0 0 0 -5px;"` even in `UITextView`.

**Comment by Chemprogrammer** [comment]
Yes, that is exactly what I want: a subscript and superscript be left-aligned. This format is very common in maths and chemistry!  Thanks again!
